de.tu_bs.cs.isf.e4cf.compare.data_structures/src/de/tu_bs/cs/isf/e4cf/compare/data_structures/io/reader/JavaNodeTypes.java

```diff
@@ -0,0 +1,28 @@
+package de.tu_bs.cs.isf.e4cf.compare.data_structures.io.reader;
+
+/**
+ * This file extends the nodes types and attributes for the generic data structure generated by @see Visitor 
+ * 
+ * @author Serkan Acar
+ * @author Pascal Blum
+ * @author Paulo Haas
+ * @author Hassan Smaoui
+ */

```

Comment:
what is with while and for ?

Reply:
We're using the class names of the nodes used in the JavaParser to set the type for these nodes.  So for "while" it is `WhileStmt`, for "for" it is `ForStmt` and `ForEachStmt` for for each loops.
Would it make sense to write our own implementation of the interface `Node` also considering [the other comment](https://github.com/TUBS-ISF/e4CompareFramework/pull/12#discussion_r560770420)?